sources/Object/Invert.ts

```diff
@@ -41,7 +41,7 @@ export type _Invert<O extends Record<Key, Key>> =
  * type test1 = O.Invert<O>
  * ```
  */
-export type Invert<O extends Record<Key, Key>> =
+export type Invert<O extends Record<keyof O, Key>> =
```

Comment:
I don't think we should use `keyof` here because it will break on unions. Is it this that caused the invert not to work on interfaces?

Reply:
Maybe it won't break, would you be able to look if we've got tests with unions on this utility. It's certainly the tests that lack the most.

Reply:
On an union of two objects with different properties basically.

Reply:
@millsp check out the new tests I just added. don't seem to break anything. The commit the removed the `key of` part was the one that caused the break...